### PR TITLE
Add flex/2.5.39

### PR DESCRIPTION
**Pull request by sourcedelica**
Specify library name and version:  **flex/2.5.39*

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
Please add `Fixes #3871` to the initial message. Thanks!

**Comment by sourcedelica** [comment]
It looks like to test clang 9.1 I would need to run a VM with Mojave or High Sierra (https://developer.apple.com/forums/thread/122508).  I'm perfectly happy to make this an invalid combination (flex/2.5.39 + apple-clang 9.1 + `flex:shared`).  How does it get removed from the CI?

**Comment by sourcedelica** [comment]
Actually I can replicate the issue with apple-clang 12.0.  Checking...

File: recipes/flex/all/conanfile.py [hunk 1]
```diff
@@ -71,7 +71,10 @@ def package(self):
             autotools = self._configure_autotools()
             autotools.install()
         tools.rmdir(os.path.join(self.package_folder, "share"))
-        os.unlink(os.path.join(self.package_folder, "lib", "libfl.la"))
+        for la in ["libfl.la", "libfl_pic.la"]:
+            la_path = os.path.join(self.package_folder, "lib", la)
+            if os.path.exists(la_path):
+                os.unlink(la_path)
```
Comment (madebr) [inline recipes/flex/all/conanfile.py:77]: You can replace these with:
```suggestion
        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
```
and put `required_conan_version = ">=1.29"` at the top of the file.

File: recipes/flex/all/conanfile.py [hunk 2]
```diff
@@ -39,10 +47,10 @@ def _configure_autotools(self):
             return self._autotools
         self._autotools = AutoToolsBuildEnvironment(self)
         configure_args = ["--disable-nls", "HELP2MAN=/bin/true", "M4=m4"]
-        if self.options.shared:
-            configure_args.extend(["--enable-shared", "--disable-static"])
-        else:
+        if self._is_static:
```
Comment (Croydon) [inline recipes/flex/all/conanfile.py:50]:
```suggestion
        if self.options.get_safe("shared"):
```

File: recipes/flex/all/conanfile.py [hunk 3]
```diff
@@ -19,6 +19,14 @@ class FlexConan(ConanFile):
...
     _autotools = None
 
+    def config_options(self):
+        if self.version == "2.5.39" and self.settings.compiler == "apple-clang":
+            del self.options.shared
+
+    @property
+    def _is_static(self):
+        return 'shared' not in self.options.values.keys() or not self.options.shared
+
```
Comment (Croydon) [inline recipes/flex/all/conanfile.py:29]:
```suggestion
```

Comment (sourcedelica) [reply recipes/flex/all/conanfile.py]: Done

File: recipes/flex/all/conanfile.py (hunk 2 above)
Comment (sourcedelica) [reply recipes/flex/all/conanfile.py]: Done

File: recipes/flex/all/conanfile.py (hunk 1 above)
Comment (sourcedelica) [reply recipes/flex/all/conanfile.py]: Done

**Comment by sourcedelica** [comment]
Question - how do we remove the combination (2.5.39, apple-clang, `flex:shared=True`) from CI?

The error happens when linking `flex`.  Short of figuring out a patch for this (there is little help for this scenario on Google) I'd prefer to just remove this configuration class.

**Pull request closed without merging**